integration/client/restart_monitor_test.go

```diff
@@ -254,6 +258,76 @@ func testRestartMonitorAlways(t *testing.T, client *Client, interval time.Durati
 	t.Logf("%v: the task was restarted since %v", time.Now(), lastCheck)
 }
 
+func testRestartMonitorPausedTaskWithAlways(t *testing.T, client *Client, interval time.Duration) {
+	if runtime.GOOS == "windows" {
+		t.Skip("Pause task is not supported on Windows")
+	}
+
+	const (
+		epsilon = 1 * time.Second
+		count   = 20
+	)
+
+	var (
+		ctx, cancel = testContext(t)
+		id          = strings.ReplaceAll(t.Name(), "/", "_")
+	)
+	defer cancel()
+
+	image, err := client.GetImage(ctx, testImage)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	container, err := client.NewContainer(ctx, id,
+		WithNewSnapshot(id, image),
+		WithNewSpec(
+			oci.WithImageConfig(image),
+			longCommand,
+		),
+		restart.WithStatus(Running),
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer func() {
+		if err := container.Delete(ctx, WithSnapshotCleanup); err != nil {
+			t.Logf("failed to delete container: %v", err)
+		}
+	}()
+
+	task, err := container.NewTask(ctx, empty())
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer func() {
+		if _, err := task.Delete(ctx, WithProcessKill); err != nil {
+			t.Logf("failed to delete task: %v", err)
+		}
+	}()
+
+	if err := task.Start(ctx); err != nil {
+		t.Fatal(err)
+	}
+
+	statusC, err := task.Wait(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	t.Log("Pause the task")
+	require.NoError(t, task.Pause(ctx))
+	defer func() {
+		require.NoError(t, task.Resume(ctx))
+	}()
+
+	select {
+	case <-statusC:
+		t.Fatal("the paused task is killed")
+	case <-time.After(30 * time.Second):
```

Comment:
Basically, the code change looks good to me. 
As far as I concerned, the test case should check the task status after 30 seconds.

Reply:
After the wait was over I added a simple status check